sdks/java/core/src/test/java/org/apache/beam/sdk/options/PipelineOptionsFactoryTest.java

```diff
@@ -753,8 +753,8 @@ public void testEmptyValueNotAllowed() {
         "--byte="};
     expectedException.expect(IllegalArgumentException.class);
     expectedException.expectMessage(
-        "Empty argument value is only allowed for String, String Array, and Collections"
-        + " of Strings");
+        "Empty argument value is only allowed for String, String Array, String ValueProvider and "
+            + "Collections of Strings");
```

Comment:
Need tests for:
ValueProvider<Collection<Integer>> and empty strings
ValueProvider<Integer[]> and empty strings
ValueProvider<Enum[]> and empty strings